app/workers/taxonomy_associations_translation_worker.rb

```diff
@@ -0,0 +1,60 @@
+class TaxonomyAssociationsTranslationWorker < WorkerBase
+  sidekiq_options queue: 'publishing_api'
+
+  def perform(model_name, id)
+    model = class_for(model_name).unscoped.find(id)
+    return if model.nil?
+
+    taxon_content_ids = taxon_content_ids_for_model(model)
+    return if taxon_content_ids.nil?
+
+    Services.publishing_api.patch_links(
+      edition.content_id,
+      links: {
+        taxons: taxon_content_ids
+      }
+    )
+  end
+
+private
+
+  def class_for(model_name)
+    model_name.constantize
+  end
+
+  def taxon_content_ids_for_model(model)
+    if model.is_a?(Edition)
+      applicable_taxon_content_ids_for_edition(model)
+    elsif model.try(:can_publish_to_publishing_api?)
```

Comment:
why try?

Reply:
Would the fallback technique work for Editions too? They definitely have link presenters - you could cut half of your code if it works :-)